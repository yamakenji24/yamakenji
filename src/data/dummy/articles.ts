import { OGPDataType } from '../../services/get-ogp';

export const dummyArticles: Array<OGPDataType> = Array(10)
  .fill(0)
  .map((_, idx) => ({
    url: '/not_found.png',
    title: 'dummy title',
    image: {
      height: '200',
      type: 'type',
      url: '/not_found.png',
      width: '200',
    }
  }))