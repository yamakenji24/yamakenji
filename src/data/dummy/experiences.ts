import { ExperienceType } from 'services/get-experience-api';

export const dummyExperiences: Array<ExperienceType> = Array(10)
  .fill(0)
  .map((_, idx) => ({
    id: idx.toString(),
    during: '',
    title: 'dummy experience',
    body: 'dummy body',
  }));
