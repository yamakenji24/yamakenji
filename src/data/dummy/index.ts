export * from './experiences';
export * from './skills';
