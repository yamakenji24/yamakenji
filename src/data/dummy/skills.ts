import { SkillType } from 'services/get-skill-api';

export const dummySkills: Array<SkillType> = Array(10)
  .fill(0)
  .map((_, idx) => ({
    id: idx.toString(),
    name: 'dummy skill',
    img: '/not_found.png',
  }));
