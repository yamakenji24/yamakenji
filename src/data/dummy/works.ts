import { WorkType } from 'services/get-work-api';

export const dummyWorks: Array<WorkType> = Array(10)
  .fill(0)
  .map((_, idx) => ({
    id: idx.toString(),
    github: '',
    link: '',
    img: '/not_found.png',
    title: 'dummy title',
    skills: [{
      fieldId: '',
      skill: '',
    }],
    body: 'dummy body',
  }))