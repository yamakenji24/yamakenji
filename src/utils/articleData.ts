export interface ArticleData {
  url: string;
  title: string;
  img: string;
  date: string;
}
export const articleData: Array<ArticleData> = [
  {
    url: 'https://zenn.dev/yamakenji24/articles/2921714d4b69d0',
    title: 'Next.jsとTailwindを利用したエンジニア名刺アプリ',
    img: '/article/cardcreator.jpg',
    date: '2020/12',
  },
  {
    url: 'https://yamakenji24.hatenablog.com/entry/2020/11/13/192543',
    title: 'Chatworkのサマーインターン2020に参加してきた',
    img: '/article/intern.jpg',
    date: '2020/09',
  },
  {
    url: 'https://shell-mag.com/vol-68/',
    title: 'シェルスクリプトマガジン Vol.68',
    img: '/article/shellvol68.jpg',
    date: '2020/08',
  },
  {
    url: 'https://qiita.com/yamakenji/items/38fd1175bdd3338fb810',
    title: 'GraphQLについてまとめてみた',
    img: '/article/GraphQL.jpg',
    date: '2020/06/23',
  },
  {
    url: 'https://qiita.com/yamakenji/items/b94fa2ab104679b29b3a',
    title: 'React+Reduxで音楽検索アプリを作ってみた',
    img: '/article/spotify.jpg',
    date: '2019/12/14',
  },
  {
    url: 'https://shell-mag.com/vol-60/',
    title: 'シェルスクリプトマガジン Vol.60',
    img: '/article/shellvol60.jpg',
    date: '2019/06',
  },
];
