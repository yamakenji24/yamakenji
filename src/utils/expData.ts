export interface ExpData {
  during: string;
  title: string;
  body: string;
}

export const expData: Array<ExpData> = [
  {
    during: 'September 14th, 2020 ~ September 18th, 2020',
    title: 'Monotaro Intern',
    body: 'インターンに参加 ~',
  },
  {
    during: 'August 17th, 2020 ~ September 4th, 2020',
    title: 'ChatWork Intern',
    body: 'インターンに参加 ~',
  },
  {
    during: 'April 2020 - present',
    title: 'Kagawa University',
    body: 'Master: Division of Reliability-based Information Systems Engineering',
  },
  {
    during: 'April 2016 ~ March 2020',
    title: 'Kagawa University',
    body: 'Bachelor: Faculty of Engineering',
  },
];
