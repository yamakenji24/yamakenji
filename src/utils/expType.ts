export interface ExperienceType {
  id: string;
  during: string;
  title: string;
  body: string;
}

export interface SkillType {
  id: string;
  name: string;
  img: string;
}
