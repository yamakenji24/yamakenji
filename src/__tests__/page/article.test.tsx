import React from 'react';
import { render } from '../utils';

import ArticleContainer from 'pages/article';
import { dummyArticles } from 'data/dummy/articles'

test('/article', () => {
  const { container } = render(
    <ArticleContainer 
      articles={dummyArticles}
      ogImageUrl={null}
    />
  )
  expect(container).toMatchSnapshot()
})