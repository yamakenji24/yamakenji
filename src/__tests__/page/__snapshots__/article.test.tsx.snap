// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/article 1`] = `
<div>
  <div
    class="css-0"
  >
    <h4
      class="chakra-heading css-1sr6c4u"
    >
      Articles
    </h4>
    <div
      class="chakra-container css-1a3ltpp"
    >
      <div
        class="css-13wc25c"
      >
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-1588jgr"
        >
          <a
            class="chakra-link css-10qsrqw"
            href="/not_found.png"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="chakra-stack css-nu2rcw"
            >
              <img
                alt="dummy title"
                class="chakra-image css-mub39v"
                src="/not_found.png"
              />
              <div
                class="css-qwpnxl"
              >
                <p
                  class="chakra-text css-0"
                >
                  dummy title
                </p>
                <p
                  class="chakra-text css-q9k0mw"
                >
                  /not_found.png
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
